libs/libcommon/src/libcommon/simple_cache.py

```diff
@@ -212,6 +212,30 @@ def get_response(kind: str, dataset: str, config: Optional[str] = None, split: O
     }
 
 
+class CacheEntryWithInfo(TypedDict):
+    dataset: str
+    config: Optional[str]
+    http_status: HTTPStatus
+    error_code: Optional[str]
+    content: Mapping[str, Any]
+
+
+def get_responses_for_kind(kind: str, dataset: str) -> List[CacheEntryWithInfo]:
```

Comment:
For this specific case I cant add the config, since I cannot send it with None. I need to get all the records in cache for an specific dataset without using the config for aggregation.  For https://github.com/huggingface/datasets-server/pull/900#discussion_r1130947540 so that we have only two calls to db.